src/orchestrator.rs

```diff
@@ -237,7 +237,13 @@ async fn detect(
         .map(|(detector_id, detector_params)| {
             let ctx = ctx.clone();
             let detector_id = detector_id.clone();
-            let detector_params = detector_params.clone();
+            let mut detector_params = detector_params.clone();
+            if let Some(default_threshold) = ctx.config.get_default_threshold(&detector_id) {
+                // Use a default threshold if threshold is not provided by the user
+                detector_params
+                    .entry("threshold".into())
+                    .or_insert(default_threshold.into());
+            }
```

Comment:
Since the orchestrator handles applying the threshold anyway, I would just add a `default_threshold` parameter to `handle_detection_task()` instead of inserting it into the `DetectorParams` map, unless there are plans for that to change.

Then, in the results `filter_map` you can do...

If we change `default_threshold` to `f32`:
```rust
  //...
  let threshold = detector_params
      .get("threshold")
      .and_then(|v| v.as_f64())
      .unwrap_or(default_threshold as f64);
  (result.score >= threshold).then_some(result)
```

Or if we keep it as `Option<f32>`:
```rust
  //...
  let threshold = detector_params
      .get("threshold")
      .and_then(|v| v.as_f64())
      .or(default_threshold.map(Into::into));
  if threshold.is_some_and(|value| result.score < value) {
      None
  } else {
      Some(result)
  }
```
(we can also get rid of the casts if we change it to `f64`)